src/java.base/share/classes/java/io/File.java

```diff
@@ -1208,10 +1214,18 @@ public File[] listFiles(FileFilter filter) {
         String[] ss = normalizedList();
         if (ss == null) return null;
         ArrayList<File> files = new ArrayList<>();
-        for (String s : ss) {
-            File f = new File(s, this);
-            if ((filter == null) || filter.accept(f))
-                files.add(f);
+        if (path.isEmpty()) {
+            for (String s : ss) {
+                File f = new File(s);
+                if ((filter == null) || filter.accept(f))
+                    files.add(f);
+            }
+        } else {
+            for (String s : ss) {
+                File f = new File(s, this);
+                if ((filter == null) || filter.accept(f))
+                    files.add(f);
```

Comment:
Rather than duplicating the loop, it could be changed to introduce `boolean isEmpty = path.isEmpty` and then the `f` can be created with `File f = isEmpty ? new File(s) : new File(this, s);`  It should be a bit cleaner.

Reply:
I had rejected that in the case of the parameter-less `listFiles` to avoid a ternary operator in each loop iteration but it _is_ cleaner. Will update.

Reply:
So changed in cfd494f.